src/lib/src/common/workspace.ts

```diff
@@ -76,6 +84,34 @@ export class Workspace {
     }
   }
 
+  observeMarker<VALUE_TYPE>(observer: MarkerObserver<VALUE_TYPE>): void {
+    if (this.pathToElement.get(observer.path)) {
+      if (!this.hasMarker(observer.path, observer.field)) {
+        this.setMarkerValue(observer.path, observer.field, null);
+      }
+      observer.observe().then((value) => this.setObservedMarker(observer, value))
+      .catch((reason) => {
+        console.log(reason);
+        this.setObservedMarker(observer, this.getMarkerValue(observer.path, observer.field));
+      });
+    } else {
+      throw new Error(`There is no element with path "${observer.path}" in this workspace.`);
+    }
+  }
+
+  private setObservedMarker<VALUE_TYPE>(observer: MarkerObserver<VALUE_TYPE>, value: VALUE_TYPE): void {
```

Comment:
I'm not convinved that 'set' is the right term here. Maybe the asynch character of the observer should be more evident. 'Set marker value and observe until stop condition is met' would be the name coming to mind first, but its is rather long. I'm not sure how to express this better in a method name.

Reply:
I went with `updateMarkerValueAndContinueObserving`. We could also consider breaking the method up, but the only reasonable way I can think of does not really solve the naming problem.